KsApi/models/graphql/adapters/Backing+BackingFragmentTests.swift

```diff
@@ -0,0 +1,534 @@
+import Apollo
+@testable import KsApi
+import XCTest
+
+final class Backing_BackingFragmentTests: XCTestCase {
+  func test() {
+    do {
+      let fragment = try GraphAPI.BackingFragment(jsonObject: backingDictionary())
+      XCTAssertNotNil(fragment)
+
+      let backing = Backing.backing(from: fragment)
+      XCTAssertNotNil(backing)
+    } catch {
+      XCTFail(error.localizedDescription)
+    }
+  }
+
+  func test_noReward() {
+    do {
+      var dict = backingDictionary()
+      dict["addOns"] = nil
+      dict["reward"] = nil
+
+      let fragment = try GraphAPI.BackingFragment(jsonObject: backingDictionary())
+      XCTAssertNotNil(fragment)
+
+      let backing = Backing.backing(from: fragment)
+      XCTAssertNotNil(backing)
+    } catch {
+      XCTFail(error.localizedDescription)
+    }
+  }
+}
+
+private func backingDictionary() -> [String: Any] {
+  let json = """
+  {
```

Comment:
How can we generate these test strings in the future?

Reply:
The way to do this is to run the query that Apollo produces in GraphiQL and copy the exact response to that query.